Issue/2086 - Refactored with Bulk Delete

mehul0810 opened the pull request:
## Description
This PR is for #2086 

## How Has This Been Tested?
I've tested this PR with multiple scenario's.
- No Donor Selected and trying delete via bulk actions.
- Single Donor Selected and trying delete via bulk actions.
- Multiple Donors Selected and trying delete via bulk actions.
- Clicking Select All Checkbox to select all donors and trying delete via bulk actions.
- Using Delete button from each donor triggering Bulk Delete feature to delete single donor as well. ( i.e. Single Delete Donor, no longer uses additional page )

## Screenshots (jpeg or gifs if applicable):
![gif](http://g.recordit.co/pEZvvnODju.gif)

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

ravinderk commented on assets/js/admin/admin-scripts.js:
```diff
@@ -1327,23 +1328,88 @@ var give_setting_edit = false;
+				var donorId   = $( this ).closest( 'tr' ).data( 'id' ),
+					donorName = $( this ).closest( 'tr' ).data( 'name' ),
+					donorHtml = '<div id="give-donor-' + donorId + '">' +
+					'<a data-id="' + donorId + '" class="give-skip-donor" title="Remove From Bulk Delete">X</a>' +
```
@mehul0810 This string should be internationalized.

ravinderk commented on includes/admin/donors/class-donor-table.php:
```diff
@@ -303,27 +303,22 @@ public function process_bulk_action() {
+				);
+				$message = give_delete_donor( $args );
+				$redirect_url = add_query_arg( 'give-message', $message, admin_url( 'edit.php?post_type=give_forms&page=give-donors' ) );
+//wp_safe_redirect( esc_url($redirect_url) );
```
@mehul0810  Remove commented code.

mehul0810 commented:
@ravinderk I've updated code as per our last discussion over call.
1. Fixed bug in select all donor, un check any donor then select all again functioning ( Tried all the possible cases)
2. Fixed Donor Delete confirmation checkbox getting checked when Selected all donors at once.
3. Smoothen the Cancel Bulk Delete option to remove the delay response.

mehul0810 commented:
@ravinderk Also, fixed the issue when there is no name for a donor under bulk delete screen. Please review with latest pull.

ravinderk commented:
@mehul0810 As we discuss overcall cancel feature is not working per donor when bulk delete tab opened.

![jaotcsiozg](https://user-images.githubusercontent.com/1784821/32763221-63191164-c924-11e7-909e-d58dd5f3f993.gif)

mehul0810 commented:
@ravinderk I've fixed and tested the issue.

![gif](http://g.recordit.co/zm7owqxTeC.gif)

ravinderk commented:
@mehul0810 Bulk delete feature is not working with the recurring addon. Donation deleted but not the subscription. I think recurring must have support for the bulk delete donor.

![screen shot 2017-11-16 at 10 15 55 am](https://user-images.githubusercontent.com/1784821/32874303-4caf6394-cab7-11e7-8606-cc8636127d33.png)



@mehul0810  Can you test deleting donor from donor edit screen also?

mehul0810 commented:
@ravinderk As discussed over call, I've fixed the positioning of Donor issue on bulk delete and the above issue is already fixed with latest release of `Recurring 1.4`.
